pkg/providers/ingress/translation/annotations/upstreamscheme/upstreamscheme.go

```diff
@@ -0,0 +1,50 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package upstreamscheme
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apache/apisix-ingress-controller/pkg/providers/ingress/translation/annotations"
+)
+
+type upstreamscheme struct{}
+
+func NewParser() annotations.IngressAnnotationsParser {
+	return &upstreamscheme{}
+}
+
+var schemeMap map[string]struct{} = map[string]struct{}{
+	"http":  {},
```

Comment:
A small suggestion. Can we make this variable global? Together with the schemes in `pkg/types/apisix/v1/types.go`.
Otherwise, we may need to deal with cross-file constant dependencies.

Reply:
good catch, we have https://github.com/apache/apisix-ingress-controller/blob/c8d3bd52fd8820475be763cd52b51b981382f285/pkg/types/apisix/v1/types.go#L52-L59

Reply:
We define SchemeMap to `pkg/types/apisix/v1/types.go`
Or just use constants like `SchemeHTTP` , `SchemeGRPC`

> pkg/types/apisix/v1/types.go

```go
var SchemeMap map[string]struct{} = map[string]struct{}{
	SchemeHTTP:  {},
	SchemeHTTPS: {},
	SchemeGRPC:  {},
	SchemeGRPCS: {},
}
```

or

> pkg/providers/ingress/translation/annotations/upstreamscheme/upstreamscheme.go

```go
var schemeMap map[string]struct{} = map[string]struct{}{
	types.SchemeHTTP:  {},
	types.SchemeHTTPS: {},
	types.SchemeGRPC:  {},
	types.SchemeGRPCS: {},
}
```